### PR TITLE
Use nn interface for the activation operator.

### PR DESCRIPTION
**Pull request by zheng-da**
This commit only contains the basic implementation of the activation operator. It has been tested on CPU and GPU. But it doesn't contain the cuDNN version of activation yet. This pull request is mainly for testing.

### CODE REVIEW

File: src/operator/nn/activation-inl.h [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+#include "../mxnet_op.h"
+#include "../operator_common.h"
+#include "../tensor/broadcast_reduce_op.h"
+
+namespace mxnet {
+namespace op {
+namespace mxnet_op {
+
+template<int req, typename Op>
+struct forward_comp {
```
Comment (piiswrong) [inline src/operator/nn/activation-inl.h]: I think we have something that does this already.

Comment (piiswrong) [reply src/operator/nn/activation-inl.h]: probably in elemwise unary/binary op

File: src/operator/nn/activation.cu [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#include "./activation-inl.h"
+#include "../tensor/elemwise_unary_op.h"
+
+namespace mxnet {
+namespace op {
+
+NNVM_REGISTER_OP(Activation)
+.set_attr<FCompute>("FCompute<gpu>", ActivationCompute<gpu>);
+
+NNVM_REGISTER_OP(_backward_activation)
```
Comment (piiswrong) [inline src/operator/nn/activation.cu]: _backward_Activation

File: src/operator/nn/activation.cu [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+.set_attr<FCompute>("FCompute<gpu>", ActivationGradCompute<gpu>);
```
Comment (piiswrong) [inline src/operator/nn/activation.cu]: We need to register FComputeEx to incorporate mkl later.
But you can also do it with FCompute for now and refactor later if you prefer

File: tests/python/unittest/test_operator.py [hunk 4]
```diff
@@ -414,6 +414,38 @@ def fsigmoid(a):
...
+    y = mx.sym.Activation(x, act_type=act_type)
+    xa = np.random.uniform(low=-1.0,high=1.0,size=shape)
+    ya = for_func(xa)
+    check_numeric_gradient(y, [xa], numeric_eps=1E-3)
+    check_symbolic_forward(y, [xa], [ya])
+    ga = back_func(ya)
+    check_symbolic_backward(y, [xa], [np.ones(shape)], [ga])
+
+def test_activation():
+    print("test activation")
```
Comment (piiswrong) [inline tests/python/unittest/test_operator.py]: remove print

File: tests/python/unittest/test_operator.py [hunk 5]
```diff
@@ -414,6 +414,38 @@ def fsigmoid(a):
     check_symbolic_backward(y, [xa], [np.ones(shape)], [ya * (1 - ya)])
 
 
+def _test_activation(act_type):
```
Comment (piiswrong) [inline tests/python/unittest/test_operator.py]: check_activation.

helper functions should start with check_

File: src/operator/nn/activation-inl.h (hunk 1 above)
Comment (zheng-da) [reply src/operator/nn/activation-inl.h]: It seems the closest thing I can find is op_with_req, but it doesn't work so well for backward computation. I'll show you a version that uses op_with_req.

**Pull request closed without merging**